Algorithm/QCAlgorithm.History.cs

```diff
@@ -399,7 +401,7 @@ public IEnumerable<T> History<T>(Symbol symbol, DateTime start, DateTime end, Re
             where T : IBaseData
         {
             var requests = CreateDateRangeHistoryRequests(new[] { symbol }, typeof(T), start, end, resolution);
-            return GetDataTypedHistory<T>(symbol, requests);
+            return GetDataTypedHistory<T>(requests, symbol).Memoize();
```

Comment:
Maybe we can push `.Memoize();` into the `GetDataTypedHistory` implementations, so all these consumers don't need to worry about it?